### ISSUE TITLE
[Feature] Use Relative Import Path in Generated Code

### ISSUE THREAD

**Issue by TheNightmareX** [ng-doc/ng-doc#146]
### Description

Currently, in the generated `ng-doc.routing.ts`, the route import paths are relative to the workspace root, which forces the workspaceRoot to serve as the tsconfig baseUrl.

ng-doc.routing.ts:
```ts
// @ts-nocheck
import {Routes} from '@angular/router';

export const NG_DOC_ROUTING: Routes = [
    {
        path: 'overview',
        loadChildren: () => import('docs/ng-doc/docs/guides/overview/module') // <-- relative to workspace root (the outDir is "docs/")
    },
];
```

In my case, the documentation application is placed under the docs/ folder and uses a separate tsconfig.json. In order to make the generated code work as expected, the tsconfig have to look like this:

tsconfig.json:
```json5
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": "../", // <-- have to use the parent folder as the baseUrl, which is wired.
    "paths": { // <-- the path mappings have to also be relative to the parent folder
      "@ng-doc/generated": ["docs/ng-doc/docs/index.ts"],
      "@ng-doc/generated/*": ["docs/ng-doc/docs/*"],
    },
  },
}
```

### Proposed solution

Since there could be other developers that hope to avoid using the workspaceRoot as baseUrl, I suggest to use import paths that relative to the current directory instead.

ng-doc.routing.ts:
```ts
// @ts-nocheck
import {Routes} from '@angular/router';

export const NG_DOC_ROUTING: Routes = [
    {
        path: 'overview',
        loadChildren: () => import('./guides/overview/module') // <-- relative to the current directory
    },
];
```

### Alternatives considered

Or maybe we could provide a new configuration option which specifieds the base path for relative imports:

ng-doc.config.js:
```js
/**@type {import('@ng-doc/builder').NgDocConfiguration} */
module.exports = {
  outDir: 'docs/',
  importBaseDir: "docs/",
};
```

By doing so, the import paths will relative to `{workspaceRoot}/docs` instead of `{workspaceRoot}`, and the developer could use a scoped tsconfig `{workspaceRoot}/docs/tsconfig.json` with the following content to make the generated code work:

tsconfig.json:
```json5
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "baseUrl": "./", // <-- now can use `./` instead of `../`
    "paths": { // <-- the path mappings are relative to the current folder now
      "@ng-doc/generated": ["ng-doc/docs/index.ts"],
      "@ng-doc/generated/*": ["ng-doc/docs/*"],
    },
  },
}
```

**Comment by TheNightmareX** [comment]
By the way, the `baseUrl` is not specified by default when generating an Angular application, and is also not added by `@ng-doc/add`. If the developer did not explicitly specify it, a TypeScript error will occur when trying to build or serve the application.

**Comment by skoropadas** [comment]
I agree. Previously, `baseUrl` was added by default when generating an Angular application, so there was no issue, but when transitioning to Vite, Angular developers also simplified the TypeScript configuration by removing `baseUrl`. Apparently, I forgot to add it during the code migration.

In general, simply adding `baseUrl: "."` as it was before should suffice, but I've also been informed (#144) that some features stopped working with the release of Angular 17.3, and I'm working on that.

Nevertheless, you're right, paths should be relative. I have already fixed this in a newer version that I'm currently working on, so it should be fixed when I release it.
